src/socket.rs

```diff
@@ -207,11 +261,59 @@ impl Socket {
     ///
     /// # Notes
     ///
-    /// Like [`Socket::new`] this will not set any flags. If that is desirable,
-    /// e.g. setting `CLOEXEC`, [`Socket::accept4`] can be used on supported
-    /// OSes or [`Socket::set_cloexec`] can be called.
+    /// On Unix-like systems, the close-on-exec flag is set on the new socket.
+    /// Additionally, on macOS X and iOS the `SOCK_NOSIGPIPE` option is set.
+    /// On Windows, the socket handle is made non-inheritable.
+    ///
+    /// If possible, the close-on-exec flag will be set atomically when the socket is created.
+    /// However, on some platforms the flag is to be set with a separate syscall,
+    /// which leaves a small window for race conditions.
+    ///
+    /// Use [`Self::accept_raw`] if you don't want these flags and options to be set.
     pub fn accept(&self) -> io::Result<(Socket, SockAddr)> {
-        sys::accept(self.inner).map(|(inner, addr)| (Socket { inner }, addr))
+        #[cfg(any(
+            target_os = "android",
+            target_os = "dragonfly",
+            target_os = "freebsd",
+            target_os = "linux",
+            target_os = "netbsd",
+            target_os = "openbsd",
+        ))]
+        {
+            self._accept4(libc::SOCK_CLOEXEC)
+        }
+
+        #[cfg(not(any(
+            target_os = "android",
+            target_os = "dragonfly",
+            target_os = "freebsd",
+            target_os = "linux",
+            target_os = "netbsd",
+            target_os = "openbsd",
+        )))]
+        {
+            let (sock, addr) = self.accept_raw()?;
+            Ok((fixup_socket(sock)?, addr))
+        }
+    }
+
+    /// Accept a new incoming connection from this listener.
+    ///
+    /// This function directly corresponds to the `accept(2)` function on
+    /// Windows and Unix.
+    ///
+    /// This function will block the calling thread until a new connection is
+    /// established. When established, the corresponding `Socket` and the
+    /// remote peer's address will be returned.
+    ///
+    /// # Notes
+    ///
+    /// This function does not set the close-on-exec flag or the `SOCK_SIGNOPIPE` option
+    /// for the created socket. On Windows, the created socket handle will be inheritable.
+    /// You should normally use [`Self::accept_raw`], which creates sockets with the advisable
+    /// flags and options for the current platform.
+    pub fn accept_raw(&self) -> io::Result<(Socket, SockAddr)> {
+        sys::accept(self.inner).map(|(inner, addr)| (Self { inner }, addr))
```

Comment:
Revert this line, not needed.

Reply:
This is an artifact of having to rebase my PR on your PR, but sure, I will.